pkg/cloud/aws/services/kubeadm/noderegistration.go

```diff
@@ -41,10 +41,12 @@ func WithTaints(taints []corev1.Taint) NodeRegistrationOption {
 	}
 }
 
-// WithCRISocket sets the location of the container runtime socket.
-func WithCRISocket(socket string) NodeRegistrationOption {
+// WithDefaultCRISocket sets the location of the container runtime socket, if it's not already set
+func WithDefaultCRISocket(socket string) NodeRegistrationOption {
```

Comment:
I think this gets a little confusing when this function may or may not update the socket, it would be more clear if it only does one thing and we figure out the crisocket before we instantiate the object

Reply:
Yeah, I didn't love the maybe-option (and the combinatorial explosion functions it represented). Your idea sounds way better, I'll do that :)